plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy

```diff
@@ -0,0 +1,85 @@
+package nextflow.slack
+
+import nextflow.plugin.Plugins
+import nextflow.plugin.TestPluginDescriptorFinder
+import nextflow.plugin.TestPluginManager
+import nextflow.plugin.extension.PluginExtensionProvider
+import org.pf4j.PluginDescriptorFinder
+import spock.lang.Requires
+import spock.lang.Shared
+import spock.lang.Timeout
+import test.Dsl2Spec
+import test.MockScriptRunner
+
+import java.nio.file.Path
+
+
+/**
+ * @author : jorge <jorge.aguilera@seqera.io>
+ *
+ */
+@Timeout(10)
+class SlackDslTest  extends Dsl2Spec{
+
+    @Shared String pluginsMode
+
+    def setup() {
+        // reset previous instances
+        PluginExtensionProvider.reset()
+        // this need to be set *before* the plugin manager class is created
+        pluginsMode = System.getProperty('pf4j.mode')
+        System.setProperty('pf4j.mode', 'dev')
+        // the plugin root should
+        def root = Path.of('.').toAbsolutePath().normalize()
+        def manager = new TestPluginManager(root){
+            @Override
+            protected PluginDescriptorFinder createPluginDescriptorFinder() {
+                return new TestPluginDescriptorFinder(){
+                    @Override
+                    protected Path getManifestPath(Path pluginPath) {
+                        return pluginPath.resolve('build/resources/main/META-INF/MANIFEST.MF')
+                    }
+                }
+            }
+        }
+        Plugins.init(root, 'dev', manager)
+    }
+
+    def cleanup() {
+        Plugins.stop()
+        PluginExtensionProvider.reset()
+        pluginsMode ? System.setProperty('pf4j.mode',pluginsMode) : System.clearProperty('pf4j.mode')
+    }
+
+    @Requires({System.getenv('SLACK_INCOMING_URL')})
```

Comment:
I wonder if mocking the HttpClient or hard coding a call to httpbin.org would be better than relying on an environment variable, here?

Reply:
Maybe you can create a similar test mocking the HttpClient to check plugin is calling the endpoint but I think can be interesting to have this and, when a valid URL is provided, review (manually) how the message appears in Slack